hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1050,6 +1051,13 @@ public boolean populateMetaFields() {
         HoodieTableConfig.POPULATE_META_FIELDS.defaultValue()));
   }
 
+  public boolean shouldMetadataExcludeKeyFromPayload() {
+    if (getBasePath() == null || !HoodieTableMetadata.isMetadataTable(getBasePath())) {
+      return false;
+    }
+    return getMetadataConfig().getBooleanOrDefault(HoodieMetadataConfig.EXCLUDE_KEY_FROM_PAYLOAD);
```

Comment:
Not a great fan of having to special case this here, at the write config level and specifically. for metadta.